src/Model/User.php

```diff
@@ -78,21 +78,105 @@ public function __construct( \WP_User $user ) {
 	 */
 	protected function is_private() {
 
+		/**
+		 * Filter whether the user is private.
+		 *
+		 * If true, the user will be considered private without further checks.
+		 *
+		 * @param mixed|null|boolean $is_private If set, return without executing the manual query
+		 *                                       to see if the user has published posts.
+		 * @param User               $this       Instance of the User Model
+		 */
+		$is_private = apply_filters( 'graphql_user_model_is_private', null, $this );
+
+		/**
+		 * If the filter returns true, we can return now as the user is considered private.
+		 *
+		 * Otherwise we need to continue and check if the User should be considered private or not
+		 */
+		if ( null !== $is_private ) {
+			return (bool) $is_private;
+		}
+
+		/**
+		 * If the requesting user does not have "list_users" capabilities and the current user is not
+		 * the user being requested, we need to check if the user being requested has published posts
+		 * which makes them a non-private entity.
+		 */
 		if ( ! current_user_can( 'list_users' ) && false === $this->owner_matches_current_user() ) {
 
 			/**
-			 * @todo: We should handle this check in a Deferred resolver. Right now it queries once per user
-			 *      but we _could_ query once for _all_ users.
-			 *
-			 *      For now, we only query if the current user doesn't have list_users, instead of querying
-			 *      for ALL users. Slightly more efficient for authenticated users at least.
+			 * Get allowed Post Types
+			 */
+			$post_types = \WPGraphQL::get_allowed_post_types();
+			unset( $post_types['revision'] );
+			unset( $post_types['attachment'] );
+
+			/**
+			 * If the user has no published posts of any allowed post type,
+			 * they are considered to be private entities.
 			 */
-			if ( ! count_user_posts( absint( $this->data->ID ), \WPGraphQL::get_allowed_post_types(), true ) ) {
-				return true;
+			if ( 0 === $this->count_user_posts( absint( $this->data->ID ), $post_types, true ) ) {
+				$is_private = true;
 			}
 		}
 
-		return false;
+		return $is_private;
+
+	}
+
+	/**
+	 * Cached version (and variation) of count_user_posts, which is uncached but doesn't always
+	 * need to hit the db
+	 *
+	 * count_user_posts is generally fast on smaller sites, but slows on large data sets.
+	 *
+	 * It can also be easy to end up with many redundant queries if it's called several times per
+	 * request. This allows bypassing the db queries in favor of the cache
+	 *
+	 * @param int @userid ID of the User
+	 * @param mixed|string|array         $post_type   The post type(s) to check for published posts in
+	 * @param boolean                    $public_only Whether to count only publicly published posts
```

Comment:
spacing is messed up here

Reply:
🤔 I used the `composer run fix-cs` command here, so it should be formatted according to the WP code standards...

Reply:
Ahh! Spacing is because of `@userid` should have been `$userid`